### DISCUSSION TITLE
PDF formdata Controls not visible in iOS Ipad

### DISCUSSION THREAD

**Discussion by GopiUC** [stephanrauh/ngx-extended-pdf-viewer#2113] [Q&A]
Hi,

I have PDF with multiple input form controls like TextBox and CheckBox, Using Chrome as a Browser.

In iOS Ipad the controls not visibile when pdf in actual size. 

![FormIssue1](https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/158054442/e61fd435-d234-41f1-a142-0fdb335cbe6a)

But I ZoomIn that pdf that controls are now visible. 

![FormIssue2](https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/158054442/6905d5ce-7270-4258-9302-b2d118735f2a)

This pdf Form working fine in PC Web and Android Tab. This issue occurred only in Ipad.

Any ideas?

Thank you for your time,

GopiVignesh

**Comment by stephanrauh** [comment]
iPad bugs are notoriosly difficult to solve... but maybe I'm lucky and it's a bug of the base viewer. What happens if you open the PDF file in https://mozilla.github.io/pdf.js/web/viewer.html?

**Reply by GopiUC** [reply]
Yeah, The controls are now visible after opening a PDF file in that viewer.

![FormIssue3](https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/158054442/7768a3c1-e504-4859-8383-09c87bc84ef3)

I need to show this controls in our Application on iPad, What need to change for this?

**Reply by stephanrauh** [reply]
Can you connect the iPad to your laptop, so you can debug it in Safari? You can look for error messages in the console window. If there's no such message, you can use the "elements" window to look for the input fields. I suspect they're there, but they don't show because of a CSS problem.

**Reply by stephanrauh** [reply]
It's also a good idea to open you file in my showcase (https://pdfviewer.net). This helps you to find out if it's a problem with your set of CSS rules, or if there's a general problem.

**Reply by stephanrauh** [reply]
It's very unlikely - but maybe updating to version 19.2.0 solves the bug. I've found some broken CSS, and maybe iPadOS couldn't cope with the bug.

**Reply by GopiUC** [reply]
Thank you for your answers and suggestions!.  I removed the [mobileFriendlyZoom] Property in my code, now its working fine in all devices.

**Reply by stephanrauh** [reply]
In theory, checkboxes should play well with `[mobileFriendlyZoom]`. Strange. However, I'm glad you've found a workaround!
